src/components/who-is-online/index.ts

```diff
@@ -0,0 +1,140 @@
+      const slot = participant.data.slotIndex;
+      const color = MeetingColorsHex[slot];
+
+      return {
+        ...participant.data,
+        participantId: participant.id,
+        color,
+      };
+    });
+
+    this.realtime.enterWhoIsOnlineChannel(this.localParticipant);
+  }
+
+  /**
+   * @function destroy
+   * @description Destroys the Who Is Online component
+   * @returns {void}
+   */
+  protected destroy(): void {}
+
+  /**
+   * @function subscribeToRealtimeEvents
+   * @description Initializes the Who Is Online component
+   * @returns {void}
+   */
+  private subscribeToRealtimeEvents(): void {
+    this.realtime.presenceSlotsInfosObserver.subscribe(this.onSlotChange);
+    this.realtime.participantJoinedObserver.subscribe(this.onParticipantJoined);
+    this.realtime.participantLeaveObserver.subscribe(this.onParticipantLeave);
+    this.realtime.whoIsOnlineObserver.subscribe(this.onWhoIsOnlineUpdate);
+  }
+
+  private onParticipantJoined = (participant): void => {
+    const alreadyJoined = this.participants?.find((element) => {
+      return element.id === participant.data.id;
+    });
+
+    if (!alreadyJoined) {
+      this.participants.push({ ...participant.data, color: this.color });
+    }
+
+    this.element.participants = this.participants;
+    this.realtime.updateWhoIsOnline(this.participants);
+  };
+
+  private onParticipantLeave = ({ data: participant }): void => {
+    this.participants = this.participants.filter((element) => {
+      return element.id !== participant.id;
+    });
+
+    this.element.participants = this.participants;
+  };
+
+  private onSlotChange = ({ id, slotIndex }): void => {
+    const color = MeetingColorsHex[slotIndex];
+    this.color = color;
+
+    this.participants = this.participants.map((participant) => {
+      if (participant.id === id) {
+        return {
+          ...participant,
+          color,
+        };
+      }
+
+      return participant;
+    });
+
+    this.realtime.updateWhoIsOnline(this.participants);
+    this.element.participants = this.participants;
+  };
+
+  private positionWhoIsOnline(): void {
+    this.element = document.createElement('superviz-who-is-online') as WhoIsOnlineElement;
+    let { location } = this;
+
+    const positionsOptions = Object.values(location);
+
+    const positionById = !positionsOptions.includes(
+      location.toLocaleLowerCase() as whoIsOnlineLocation,
+    );
+
+    if (positionById) {
+      const container = window.document.body.querySelector(`#${location}`);
+
+      if (container) {
+        container.appendChild(this.element);
+        this.element.position = 'position: relative;';
+        return;
+      }
+
+      location = LOCATION.TOP_RIGHT;
+    }
+
+    document.body.appendChild(this.element);
+    const [vertical, horizontal] = location.split('-');
+    this.element.position = `${vertical}: 20px; ${horizontal}: 20px;`;
+  }
```

Comment:
It's more clean, because: 

1. The default position is the case is more usable.
2. If the container doesn't exist, use the default position. The skip first principle
3. Remove the nested if's

```suggestion
  private positionWhoIsOnline(): void {
    this.element = document.createElement('superviz-who-is-online') as WhoIsOnlineElement;

    const isUsingDefaultPosition = Object.values(this.location).includes(
      location.toLocaleLowerCase() as whoIsOnlineLocation,
    );
    
    if(isUsingDefaultPosition) {
      document.body.appendChild(this.element);
      const [vertical, horizontal] = this.location.split('-');
      this.element.position = `${vertical}: 20px; ${horizontal}: 20px;`;
      return;
    }
    
      const container = document.getElemetById(this.location);

      if (!container) {
         location = LOCATION.TOP_RIGHT;    
          return;
        }
   
      container.appendChild(this.element);
      this.element.style.position = 'relative'
    }
  }
```

Another suggestion is: if the container is anything other than static, don't change the position. Because it may break the client layout.